File: 백준/Silver/11345. Cornell Party/Cornell Party.py
import sys

def classify(N, ids):
    count = [0] * 1000001
    for id in ids:
        count[id] += 1
    ans = [0] * (N + 1)
    for c in count:
        if c != 0:
            ans[c] += 1
    return ans

def main():
    input = sys.stdin.read  # Read all input at once
    data = input().split()  # Split by whitespace to get all numbers in a list
    
    index = 0
    T = int(data[index])
    index += 1
    
    results = []
    
    for _ in range(T):
        N = int(data[index])
        index += 1
        
        cornell_ids = list(map(int, data[index:index + N]))
        index += N
        
        white_ids = list(map(int, data[index:index + N]))
        index += N
        
        cornell = classify(N, cornell_ids)
        white = classify(N, white_ids)
        
        if cornell == white:
            results.append("what a lovely party")
        else:
            results.append("you've messed up, Cornell")
    
    # Output all results at once to avoid multiple IO operations
    sys.stdout.write("\n".join(results) + "\n")

if __name__ == "__main__":
    main()

File: 백준/Silver/11345. Cornell Party/Cornell Party.java
import java.util.Scanner;

class Main { // 클래스 이름을 Solution에서 Main으로 변경
    private static int[] classifyGuestsByIdFrequency(int numberOfGuests, Scanner scanner) {
        int[] idFrequency = new int[1000001];
        for (int i = 0; i < numberOfGuests; ++i) {
            idFrequency[scanner.nextInt()]++;
        }
        int[] frequencyCount = new int[numberOfGuests + 1];
        for (int i = 0; i < 1000001; ++i) {
            if (idFrequency[i] != 0) frequencyCount[idFrequency[i]]++;
        }
        return frequencyCount;
    }
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int numberOfTestCases = scanner.nextInt();
        for (int testCaseIndex = 0; testCaseIndex < numberOfTestCases; ++testCaseIndex) {
            int numberOfGuests = scanner.nextInt();
            int[] cornellGuestFrequency = classifyGuestsByIdFrequency(numberOfGuests, scanner);
            int[] whiteGuestFrequency = classifyGuestsByIdFrequency(numberOfGuests, scanner);
            boolean frequenciesMatch = true;
            for (int i = 0; i < numberOfGuests + 1; ++i) {
                if (cornellGuestFrequency[i] != whiteGuestFrequency[i]) frequenciesMatch = false;
            }
            if (frequenciesMatch) System.out.println("what a lovely party");
            else System.out.println("you've messed up, Cornell");
        }
    }
}